shared/crypto/slip/slip_test.go

```diff
@@ -0,0 +1,228 @@
+			seed:        testSeedHex,
+			wantPrivHex: "849e1cf075e4ba4a310c2f89036a2a7eb2bb18ccfcfdd793cdc48a28978e588e",
+			wantPubHex:  "009ab1c51476edbb1f71f91b33d3bb342503cba9adc8cd4a4c6e62a8cc7d27859b",
+			wantErr:     false,
+		},
+		{
+			name:        "PoktTestVector Key derivation is deterministic for path `m/44'`",
+			path:        "m/44'",
+			seed:        testSeedHex,
+			wantPrivHex: "35852227a2289bcba3f0a428e08cc864f2ed35f605552c378232856a2abf5564",
+			wantPubHex:  "00d0b243e9de9f2f12410275025395bd088cf57b0272794cc66d66093a51579797",
+			wantErr:     false,
+		},
+		{
+			name:        "PoktTestVector Key derivation is deterministic for path `m/44'/635'`",
+			path:        "m/44'/635'",
+			seed:        testSeedHex,
+			wantPrivHex: "9af532407b4dba729962f9d57496394e7b82ea7a995575d0bfcc8fd4845debe6",
+			wantPubHex:  "00a428992b7afb7716cf339d2b54bede7f2932cf4cf3c542f5662dcebd7e3abaee",
+			wantErr:     false,
+		},
+		{
+			name:        "PoktTestVector Child key derivation is deterministic for index `0` (first child)",
+			path:        fmt.Sprintf(PoktAccountPathFormat, 0),
+			seed:        testSeedHex,
+			wantPrivHex: "e7e0f734311bdf2f446821a464b490ae8c005f6af92a26f42a39067103726346",
+			wantPubHex:  "000848875b5836bc71fecb10f563b84b9b8a63b6d7a1b16b1e88eeaca6a7ad5852",
+			wantErr:     false,
+		},
+		{
+			name:        "PoktTestVector Child key derivation is deterministic for index `1000000`",
+			path:        fmt.Sprintf(PoktAccountPathFormat, 1000000),
+			seed:        testSeedHex,
+			wantPrivHex: "f208bf7d7afa1a12a0b74e8fbbf1bb15bdfcaa3d03507ec338d7ecd77331e964",
+			wantPubHex:  "00e7d8f01dfbc0eb1638d9853b95cdec650ad47d72fac1d2d1c97c8f935d2cbf90",
+			wantErr:     false,
+		},
+		{
+			name:        "PoktTestVector Child key derivation is deterministic for index `2147483647` (last child)",
+			path:        fmt.Sprintf(PoktAccountPathFormat, 2147483647),
+			seed:        testSeedHex,
+			wantPrivHex: "20e061dcfae5cc90cba8ee374afc2d67d518b5f542f3217b8c830293c3dbb7e6",
+			wantPubHex:  "0008a399a3cdc1ee9a50c922d018daa98b5069bfea37944fde14a73d83ca3ec08c",
+			wantErr:     false,
+		},
+		{
+			name:        "PoktTestVector Child index is too large to derive ed25519 key for index `2147483648` ",
+			path:        fmt.Sprintf(PoktAccountPathFormat, 2147483648),
+			seed:        testSeedHex,
+			wantPrivHex: "",
+			wantPubHex:  "",
+			wantErr:     true,
+		},
+		{
+			name:        "PoktTestVector Child index is too large to derive ed25519 key for index `4294967295` ",
+			path:        fmt.Sprintf(PoktAccountPathFormat, ^uint32(0)),
+			seed:        testSeedHex,
+			wantPrivHex: "",
+			wantPubHex:  "",
+			wantErr:     true,
+		},
+	}
+	for _, tv := range tests {
+		t.Run(tv.name, func(t *testing.T) {
+			seed, err := hex.DecodeString(tv.seed)
+			require.NoError(t, err)
+			childKey, err := DeriveChild(tv.path, seed)
+			if tv.wantErr {
+				require.Error(t, err)
+			} else {
+				require.NoError(t, err)
+			}
+
+			if err != nil {
```

Comment:
Do we need this if structure?

Reply:
Yeah this makes sure that if the error is present (for the tests when we want it) it doesn't try to do the rest of the test and throw null pointer exceptions because the other stuff isn't there.

Reply:
Adding a suggestion, should it be a `continue` then so we run the remainder of the tests?

Reply:
@Olshansk I don't think `continue` would work as this is not in a for loop. From my understanding the `t.Run` will run all of the test cases in separate "loops" so the return is only blocking the error cases from failing the tests it shouldn't pass anyway. This return doesn't block the other tests from running